project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php

```diff
@@ -48,32 +48,31 @@ public function testGetArticleReturnsError(): void
         $this->assertSame($expectedErrorMessage, $errors[0]['message']);
     }
 
-    /**
-     * @param string $graphQlType
-     * @param array $expectedData
-     * @dataProvider getSpecialArticleDataProvider
-     */
-    public function testGetSpecialArticle(string $graphQlType, array $expectedData): void
+    public function testGetSpecialArticle(): void
     {
-        $response = $this->getResponseContentForQuery($this->getSpecialArticleQuery($graphQlType));
-        $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
-        $responseData = $this->getResponseDataForGraphQlType($response, $graphQlType);
+        foreach ($this->getSpecialArticleDataProvider() as $dataSet) {
```

Comment:
why remove @dataprovider and wrap the whole method into foreach?

Reply:
The dataprovider is dependent on property set via `setUp` method. Sadly phpunit uses dataprovider sooner than `setUp` is called so it end on `calling * on null`

Reply:
Sad, but I understand.